platform/admin/core/admin-core-api/src/test/java/org/codice/ddf/ui/admin/api/SystemPropertiesAdminTest.java

```diff
@@ -0,0 +1,241 @@
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.ui.admin.api;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.Matchers.equalTo;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+
+import javax.management.MalformedObjectNameException;
+
+import org.codice.ddf.configuration.SystemBaseUrl;
+import org.codice.ddf.configuration.SystemInfo;
+import org.codice.ddf.security.certificate.generator.CertificateGeneratorMBean;
+import org.codice.ddf.security.certificate.keystore.editor.KeystoreEditor;
+import org.codice.ddf.security.certificate.keystore.editor.KeystoreEditorMBean;
+import org.codice.ddf.ui.admin.api.impl.SystemPropertiesAdmin;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+public class SystemPropertiesAdminTest {
+    File etcFolder = null;
+
+    File systemPropsFile = null;
+
+    File userPropsFile = null;
+
+    SystemBaseUrl sbu = null;
+
+    SystemInfo info = null;
+
+    @Before
+    public void setUp() throws IOException {
+        System.setProperty(SystemBaseUrl.HOST, "host");
+        System.setProperty(SystemBaseUrl.PORT, "1234");
+        System.setProperty(SystemBaseUrl.HTTP_PORT, "4567");
+        System.setProperty(SystemBaseUrl.HTTPS_PORT, "8901");
+        System.setProperty(SystemBaseUrl.PROTOCOL, "https://");
+        System.setProperty(SystemInfo.ORGANIZATION, "org");
+        System.setProperty(SystemInfo.SITE_CONTACT, "contact");
+        System.setProperty(SystemInfo.SITE_NAME, "site");
+        System.setProperty(SystemInfo.VERSION, "version");
+
+        TemporaryFolder temporaryFolder = new TemporaryFolder();
+        temporaryFolder.create();
+        etcFolder = temporaryFolder.newFolder("etc");
+        System.setProperty("karaf.etc", etcFolder.getAbsolutePath());
+
+        systemPropsFile = new File(etcFolder, "system.properties");
+        userPropsFile = new File(etcFolder, "users.properties");
+
+        sbu = new SystemBaseUrl();
+        info = new SystemInfo();
+    }
+
+    @Test
+    public void testReadSystemProperties() {
+        SystemPropertiesAdmin spa = new SystemPropertiesAdmin();
+        List<SystemPropertyDetails> details = spa.readSystemProperties();
+        assertThat(details.size(), is(9));
```

Comment:
Should we rename this or adjust the assertion to really test that the system properties are being read correctly? 
If I understand this correctly it's just asserting that 9 properties were read